test/parallel/test-tls-lookup.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const tls = require('tls');
+
+const expectedError = /^TypeError: "lookup" option should be a function$/;
+
+['foobar', 1, {}, []].forEach((input) => connectThrows(input));
+
+function connectThrows(input) {
+  const opts = {
+    host: 'localhost',
+    port: common.PORT,
+    lookup: input
+  };
+
+  assert.throws(function() {
+    tls.connect(opts);
+  }, expectedError);
+}
+
+connectDoesNotThrow(common.mustCall(common.noop));
```

Comment:
@lpinca You mean don't use `common.noop` here and use `() => {}` instead? (If so: :+1:)